### PR TITLE
Move validation to rjsf / ajv validator

### PR DESCRIPTION
**Pull request by marthacryan**
Fixes elyra-ai/elyra#2933. Makes the validation consistent between rjsf and the pipeline-services package. 

Todo:
- [x] Shows validation error indicator for entire form
- [x] Tooltip broken for some nodes
- [x] Some extra validation indicators
- [x] Issue with switching oneOf option
- [x] Fix pipeline parameters validation error float says "Should be bool"
- [ ] Extra validation errors in tooltip
 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: packages/pipeline-editor/src/CustomFormControls/CustomFieldTemplate.tsx [hunk 1]
```diff
@@ -14,22 +14,21 @@
...
+  if (props.uiSchema?.["ui:readonly"]) {
     children = (
       <div style={{ paddingTop: "8px" }}>
         {props.formData ?? props.schema.default}
       </div>
     );
   }
-  const requiredError = props.required && props.formData === undefined;
-  const hasError = props.rawErrors || requiredError;
+  const hasError = (props.rawErrors?.length ?? 0) > 0;
```
Comment (ajbozarth) [inline packages/pipeline-editor/src/CustomFormControls/CustomFieldTemplate.tsx:31]: would `const hasError = !!props.rawErrors?.length` give the same result in simpler code?

**Review by ajbozarth — COMMENTED** [review]
On first pass code review only 1 comment so far, still need to detail read the actual validation code and test it locally.

I am hesitant to depend on a beta release though, we probably should wait until the final release before we commit this

File: packages/pipeline-editor/src/CustomFormControls/CustomFieldTemplate.tsx (hunk 1 above)
Comment (marthacryan) [reply packages/pipeline-editor/src/CustomFormControls/CustomFieldTemplate.tsx:31]: Would that check for rawErrors having length 0?

Comment (marthacryan) [reply packages/pipeline-editor/src/CustomFormControls/CustomFieldTemplate.tsx:31]: Ohh wait ok I think I see what you're saying

Comment (ajbozarth) [reply packages/pipeline-editor/src/CustomFormControls/CustomFieldTemplate.tsx:31]: it should work since 0 is falsy

**Comment by marthacryan** [comment]
Since rjsf 5 releases are still in beta, we're going to wait on the official release to merge this.

**Comment by ptitzler** [comment]
Upstream issue:

```
https://github.com/rjsf-team/react-jsonschema-form/issues/3144
```

**Comment by ptitzler** [comment]
No longer blocked as [RJSF 5.x was released](https://github.com/rjsf-team/react-jsonschema-form/releases)

**Pull request closed without merging**